src/js/legend_control.js

```diff
@@ -686,8 +686,9 @@ export default class LegendControl {
     if (!networkMetadata) return;
 
     let section = document.getElementById("legend-section-shields");
```

Comment:
We could pull this statement up to the top of the method and short-circuit fetching the metadata from Wikidata. But even if that query runs unnecessarily the first time the user opens the legend, it avoids a flash of raw tag values when opening the legend again in the presence of shields. I think this is a better user experience worth the cost of a relatively small request.